.github/workflows/java-publish.yml

```diff
@@ -1,9 +1,8 @@
 name: Build and publish Java packages
 on:
   release:
-    # Use released instead of published, since we don't publish preview/beta
-    # versions
-    types: [released]
+    # Trigger on published to include both stable and preview/beta releases
+    types: [published]
```

Comment:
since Maven Central does not have hard limit, and we bump version also for preview release now, we can publish java all the time